pandas/core/dtypes/missing.py

```diff
@@ -391,43 +399,28 @@ def array_equivalent(left, right, strict_nan: bool = False) -> bool:
     if left.shape != right.shape:
         return False
 
+    if dtype_equal:
```

Comment:
i am not sure i understand why you actually need this
the dtype check is cheap compared to the actual check no?

Reply:
1. They add up, especially when doing things columnwise for small, wide dataframes
2. We get to skip a few other checks, like swapping `np.isnan` for `pd.isna` for float, and we can skip the empty check at https://github.com/pandas-dev/pandas/pull/35328/files#diff-ff8364cee9a3e1ef3a3825cb2cdd26d8L431.

Reply:
ok that's fine